### PR TITLE
Magefile and Localdev merges - first run

### PR DESCRIPTION
**Pull request by Sharpz7**
# Summary

Closes https://github.com/armadaproject/armada/issues/2221
Closes https://github.com/armadaproject/armada/issues/2235
Closes https://github.com/armadaproject/armada/issues/2246
Closes https://github.com/armadaproject/armada/issues/2247
Closes https://github.com/armadaproject/armada/issues/2248

# The New Process

Instead of compiling from docker-compose, localdev now builds the CI docker images locally. This doesn't slow us down (We still have to compile the services, lookoutUI anyway), and instead encourages users to use good, supported debugging methods (Delve, VSCode, Goland)

# How to Use

For quick testing, run `developer/init.sh` for a quick build, but you can also still use the individual mage targets and docker-compose commands.

Please refer to: 
- https://github.com/armadaproject/armada/blob/d53cff9f9e33026a64372c2539a841f57dab4402/docs/developer.md
- https://github.com/armadaproject/armada/blob/d53cff9f9e33026a64372c2539a841f57dab4402/docs/development_guide.md

Also note that `localdev/run.sh` still works :))

# Problems

## Delay before Armada Runs

There seems to be a slight delay between Armada Booting, and the testsuite successfully running. ~(30seconds)
Before this time, you get:

**Update**
This seems to be a known design problem: https://github.com/armadaproject/armada/blob/master/magefiles/ci.go#L30. I have therefore added a 15 second pause.

# Breaking Changes

Delve debugging has not been tested, extensively, so expect problems until https://github.com/armadaproject/armada/issues/2249

`go-path` also seemed to cause problems in the debug docker-compose `dlv` would fail to be found.

### CODE REVIEW

File: developer/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Dependencies (TEMPORARY)
```
Comment (kannon92) [inline developer/README.md:1]: Why Temporary?

File: localdev/config/armada/config.yaml [hunk 2]
```diff
@@ -1,17 +1,13 @@
...
-priorityHalfTime: 20m
-cancelJobsBatchSize: 1000
 grpc:
   keepaliveParams:
     maxConnectionIdle: 5m
     time: 2h
     timeout: 20s
   keepaliveEnforcementPolicy:
     minTime: 10s
+    # ?
```
Comment (kannon92) [inline localdev/config/armada/config.yaml:8]: Why is this here?

File: localdev/config/executor/config.yaml [hunk 3]
```diff
@@ -1,42 +1,27 @@
...
+  # This line is different
   impersonateUsers: true
   etcd:
     fractionOfStorageInUseSoftLimit: 0.8
     fractionOfStorageInUseHardLimit: 0.9
     minimumAvailable: 2
+  # These lines are different
   QPS: 100
   Burst: 100
+
```
Comment (kannon92) [inline localdev/config/executor/config.yaml:17]:
```suggestion
```

**Review by kannon92 — DISMISSED** [review]
Nits but it lgtm

File: developer/README.md (hunk 1 above)
Comment (Sharpz7) [reply developer/README.md:1]: Just leaving this here for reference - I will need to re-write the dev docs.

File: config/executor/config.yaml [hunk 4]
```diff
@@ -19,6 +19,7 @@ task:
   utilisationEventReportingInterval: 5m
 apiConnection:
   armadaUrl: "server:50051"
+  forceNoTls: false
```
Comment (Sharpz7) [inline config/executor/config.yaml:22]: I am assuming this will not break anything?

It is required to allow the env var to work I think, it can only replace, not create.

File: developer/env/armada.env [hunk 5]
```diff
@@ -0,0 +1 @@
+ARMADA_SCHEDULING_PREEMPTION_DEFAULTPRIORITYCLASS=""
```
Comment (Sharpz7) [inline developer/env/armada.env:1]: Created seperate env files to stop future potential conflicts.

i.e `ARMADA_POSTGRES_CONNECTION_DBNAME` for lookout or lookoutv2? there is no distinction.

File: magefiles/developer.go [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	mg.Deps(StopComponents)
+	mg.Deps(StopDependencies)
+	mg.Deps(KindTeardown)
+}
+
+// Dependencies include pulsar, postgres (v1 and v2) as well as redis
+func StartDependencies() error {
+	// If user is on arm, export to env "PULSAR_IMAGE=kezhenxu94/pulsar"
+	if onArm() {
+		os.Setenv("PULSAR_IMAGE", "kezhenxu94/pulsar")
```
Comment (kannon92) [inline magefiles/developer.go:40]: Prob can use Rich’s image

Comment (Sharpz7) [reply magefiles/developer.go:18]: Good point! Thank you.

**Review by kannon92 — APPROVED** [review]
